### PR TITLE
SAK-50427 LTI Enhance the UI for Tool Deployment Feature

### PR DESCRIPTION
**Pull request by zhuoY121**
https://sakaiproject.atlassian.net/browse/SAK-50427

### CODE REVIEW

**Review by csev — APPROVED** [review]
LGTM

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        </p>
+    </form>
+
+    #if (($isEdit && $isAdmin && $autoStartUrl) || $canDeploy)
+        <div id="modal-iframe-div" style="display:none" tabindex="-1" role="dialog">
+            <iframe name="sakai-basiclti-admin-iframe" id="sakai-basiclti-admin-iframe" src="/library/image/sakai/spinner.gif" style="min-height: 80vh;" tabindex="0"></iframe>
+        </div>
+
+        <script>
+        //<![CDATA[
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm:46]: looks like you can remove this?

**Review by ern — APPROVED** [review]

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+        includeLatestJQuery('lti_tool_edit.vm');
+        includeWebjarLibrary('fontawesome-iconpicker');
+    </script>
+    <h3>$tlang.getString("tool.put")</h3>
+
+    #if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $formattedText.escapeHtml($alertMessage)</div>#end
+
+    <form action="#toolForm("")" method="post" name="customizeForm" >
+        #if ($isEdit && $isAdmin && $autoStartUrl )
+            <div style="float:right" id="lti-advantage-div">
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm:13]: avoid inline style if possible?

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+                <a href="$autoStartUrl" class="btn btn-primary" target="sakai-basiclti-admin-iframe" title="$tlang.getString("tool.lti13.auto.title")"
+                   role="button" onclick="showIframe(this.title, false);">$tlang.getString("tool.lti13.auto.button")
+                </a>
+            </div>
+        #end
+
+        #if ($canDeploy)
+            <div style="float:right; display: none;" id="lti-deployment-div">
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm:21]: avoid inline style if possible?

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm [hunk 4]
```diff
@@ -162,27 +161,32 @@ ${includeLatestJQuery}
...
 	                #if ($isAdmin) <td headers="toolorder">$!tool.get("toolorder")</a></td> #end
 	                #if ( $isAdmin || $allowMaintainerAddSystemTool) 
 		                <td headers="action">
 		                	#if ( $isAdmin || ($allowMaintainerAddSystemTool && $getContext == $tool.get("SITE_ID")))
                                 <ul style="list-style: none;">
-	                        	<li style="display:inline"><a href="$sakai_ActionURL.setPanel("ToolEdit")&id=$tool.get("id")">$tlang.getString("gen.edit")</a></li>
-									#if (!$tool.get("SITE_ID") && $tool.get("visible") == 1)
-										## If SITE_ID is blank and the tool is stealthed, then it can be deployed
-										<li style="display:inline"><span aria-hidden="true"> / </span> <a href="$sakai_ActionURL.setPanel("ToolSiteDeploy")&tool_id=$tool.get("id")">$tlang.getString("gen.deploy")</a> </li>
+									<li style="display:inline"><a href="$sakai_ActionURL.setPanel("ToolEdit")&id=$tool.get("id")">$tlang.getString("gen.edit")</a></li>
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm:176]: avoid inline style if possible?

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm [hunk 5]
```diff
@@ -109,7 +109,11 @@ ${includeLatestJQuery}
 	#if ($messageSuccess)<div class="sak-banner-success">$tlang.getString("gen.success") $formattedText.escapeHtml($messageSuccess)</div>#end
 	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $formattedText.escapeHtml($alertMessage)</div>#end
         #if ($configMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $formattedText.escapeHtml($configMessage)</div>#end
-	#if ($isAdmin || $allowMaintainerAddSystemTool)<div align="right" style="padding:10px"><a href="" title="$tlang.getString("add.to.system")" onclick="location = '$sakai_ActionURL.setPanel("ToolInsert")';return false;"> $tlang.getString("add.to.system")</a>#if ($isAdmin )<br/><a href="" title="$tlang.getString("add.auto")" onclick="location = '$sakai_ActionURL.setPanel("AutoInsert")';return false;"> $tlang.getString("add.auto")</a>#end</div>#end
+	#if ($isAdmin || $allowMaintainerAddSystemTool)
+		<div align="right" style="padding:10px">
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm:113]: inline syle

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm [hunk 6]
```diff
@@ -1,359 +1,42 @@
...
+
+        <div id="title-input" class="foorm-text" style="clear:both;">
+            <label for="title" style="display:block;float:none;font-weight:bold"><span class="foorm-required" style="color:#903;font-weight:bold;" title="$tlang.getString('tool.new.insert.start.title')">*</span>$tlang.getString('tool.new.insert.start.title')</label>
+            <div id="div_title">
+                <input type="text" id="title" name="title" size="40" style="border:1px solid #555;padding:5px;font-size:1em;width:300px" value="$!previousPost.get("title")"/>
+            </div>
+        </div>
+
+        #if ($isAdmin)
+            <div id="siteId-input" class="foorm-text" style="clear:both;">
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm:23]: inline style

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm [hunk 7]
```diff
@@ -1,359 +1,42 @@
...
+                <label for="siteId" style="display:block;float:none;font-weight:bold">$tlang.getString('tool.new.insert.start.siteId')</label>
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm:24]: inline style

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm [hunk 8]
```diff
@@ -1,359 +1,42 @@
...
+                <div id="div_siteId"><input type="text" id="siteId" name="siteId" size="40" style="border:1px solid #555;padding:5px;font-size:1em;width:300px" value="$!previousPost.get("siteId")"/></div>
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm:25]: inline style

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm [hunk 9]
```diff
@@ -1,359 +1,42 @@
...
-#if ($isEdit && $isAdmin && $autoStartUrl )
-        <div style="float:right" id="lti-advantage-div"><a href="$autoStartUrl" class="btn btn-primary" target="sakai-basiclti-admin-iframe"
-        title="$tlang.getString("tool.lti13.auto.title")" role="button"
-        onclick="showIframe(this.title, false);">$tlang.getString("tool.lti13.auto.button")</a></div>
-#end
-        $formInput
-                <input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
+        <input type="hidden" name="sakai_csrf_token" value="$sakai_csrf_token" />
+
+        <div id="title-input" class="foorm-text" style="clear:both;">
```
Comment (ern) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_insert.vm:15]: inline style

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm (hunk 2 above)
Comment (csev) [reply basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm:13]: I don't think so - This is not a general thing - it is just one button that needs to float right.  Any other approach would be harder to understand for future developers.

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm (hunk 1 above)
Comment (csev) [reply basiclti/basiclti-tool/src/webapp/vm/lti_tool_edit.vm:46]: Are you saying that the //<![CDATA[ is no longer needed for page validation?

**Pull request merged**